tests/e2e/run-compatibility.sh

```diff
@@ -84,6 +87,17 @@ _run_compatibility_tests() {
     collect_and_check_stackrox_logs "/tmp/compatibility-test-logs" "${compatibility_dir}/initial_tests"
 }
 
+update_junit_prefix_with_central_and_sensor_version() {
+    local short_central_tag="$1"
+    local short_sensor_tag="$2"
+
+    result_folder="${ROOT}/tests/compatibility-tests-results"
+    info "Updating all test in $result_folder to have \"Central-v${short_central_tag}-Sensor-v${short_sensor_tag}_\" prefix"
+    find "${result_folder}" -type f -name "*.xml" -exec sh -c '
+            sed -i "s/testcase name=\"/testcase name=\"[Central-v${short_central_tag}-Sensor-v${short_sensor_tag}] /g" "$f"
+    ' sh {} \;
```

Comment:
I am afraid that won't work because `$f` is not existing. It looks pretty complex - it calls `sh` shell twice. Can you think of something simpler maybe? What do you think of [this answer](https://stackoverflow.com/a/37210472)? It explains how to fix the variant with `for` and glob and also presents solution using `find` that is much easier to read.

Reply:
Right, it should be "$1" instead of "$f", slipped over it while changing the loop to a find.

I was reading about the find command yesterday, here's a quite interesting [stackexchange thread](https://unix.stackexchange.com/questions/389705/understanding-the-exec-option-of-find) about it. Basically from how I understand it it's the more correct way to -exec over the items found with find rather than looping over them. The first `sh -c` calls a subshell within which I can then run commands, the second `sh` passes.

I think `-exec` is the correct way to iterate over objects found with `find` efficiently, but let me know if you disagree. For now I'll run the tests again and make sure this works properly.

Reply:
I read the linked article and could understand why we use two `sh` as follows:

- The first `sh -c` defines a command to run. The command takes parameter `$1`.
- The second `sh` executes the command and passes the `find` result `{}` into `$1` of the command.

Reply:
Okay fine, I'll fall back to the `for` loop + `glob` option.

I got excited to use the fancy `find` options I had read about, but you're right that readability is more important. I've added the `continue` if no file is found as suggested in the stackoverflow post you linked.

Reply:
The current code will work for `sh` and `bash`. It prints extra error in `zsh`. This is totally fine.
You mention using the `glob` option, but the code has none - do you still plan to add it? I think, the code can stay as it is.

No need to get demoted to not use `find` - it is a very nice tool, however, I would rather prefer using less tricky variants of the command (even if they are not oneliners), so that they are easy to understand and change later.